snapcraft/plugins/python.py

```diff
@@ -365,14 +213,100 @@ def build(self):
             for name in installed_pipy_packages
         ]
 
-        self._fix_permissions()
+        _python.generate_sitecustomize(
+            self._python_major_version, stage_dir=self.project.stage_dir,
+            install_dir=self.installdir)
+
+    def _get_setup_py_dir(self):
+        setup_py_dir = None
+        setup_py = 'setup.py'
+        if os.listdir(self.sourcedir):
+            setup_py = os.path.join(self.sourcedir, 'setup.py')
+        if os.path.exists(setup_py):
+            setup_py_dir = os.path.dirname(setup_py)
+
+        return setup_py_dir
 
-        # Fix all shebangs to use the in-snap python.
-        file_utils.replace_in_file(self.installdir, re.compile(r''),
-                                   re.compile(r'^#!.*python'),
-                                   r'#!/usr/bin/env python')
+    def _get_constraints(self):
+        constraints = None
+        if self.options.constraints:
+            if isurl(self.options.constraints):
+                constraints = {self.options.constraints}
+            else:
+                constraints = {os.path.join(
+                    self.sourcedir, self.options.constraints)}
+        return constraints
+
+    def _get_requirements(self):
+        requirements = None
+        if self.options.requirements:
+            if isurl(self.options.requirements):
+                requirements = {self.options.requirements}
+            else:
+                requirements = {os.path.join(
+                    self.sourcedir, self.options.requirements)}
+
+        return requirements
+
+    def _install_wheels(self, wheels):
+        installed = self._pip.list()
+        wheel_names = [os.path.basename(w).split('-')[0]
+                       for w in wheels]
+
+        # we want to avoid installing what is already provided in
+        # stage-packages
+        need_install = [k for k in wheel_names if k not in installed]
+        self._pip.install(
+            need_install + self.options.python_packages, upgrade=True,
+            install_deps=False,
+            process_dependency_links=self.options.process_dependency_links)
+
+    def _download_from_setup_py(self):
```

Comment:
I am confused by this method name, should it not be something closer to `download_project`?

Reply:
Haha, I obviously got into the weeds with that. `{download,install}_project` is obviously a far better choice, thank you. Done.